controllers/course.js

```diff
@@ -102,42 +102,65 @@ const updateVideoStatus = async (req, res = response) => {
   }
 };
 
-const lastViewedVideo = async (req, res = response) => {
+const lastViewedVideos = async (req, res = response) => {
   const { id } = req.params;
   const { courseName, courseId, videoId, tema, indexTopic, urlVideo } =
     req.body;
 
   try {
-    const user = await User.findOne({ _id: id });
-
+    let user = await User.findOne({ _id: id });
     if (!user) {
       return res.status(404).json({
         ok: false,
         msg: "Usuario no encontrado",
       });
     }
 
-    (user.lastViewedInfo = {
-      courseName,
-      idCourse: courseId,
-      idVideo: videoId,
-      tema,
-      indexTopic,
-      urlVideo,
-    }),
-      await user.save();
+    if (!user.lastViewedVideos) {
+      user.lastViewedVideos = [];
+    }
+
+    let courseIndex = user.lastViewedVideos.findIndex(
+      (info) => info.idCourse === courseId
+    );
+
+    if (courseIndex !== -1) {
+      user.lastViewedVideos[courseIndex] = {
+        courseName,
+        idCourse: courseId,
+        idVideo: videoId,
+        tema,
```

Comment:
Should this be in English?

Reply:
Yes Sthefa should be in English but that information comes from the Front end and it is necessary to make changes in the repositories where the properties are in Spanish

Reply:
Yes Sthefa must be in English but that information comes from the Front-end and it is necessary to make changes in the repositories where the properties are in Spanish and for that reason for the moment I can not change it to English any property that is received by params or body @SthefaniaEcheverry @DanielCalleCO